src/router/index.ts

```diff
@@ -170,58 +204,84 @@ const routes: Array<RouteRecordRaw> = [
         path: 'accounts',
         name: 'ProfileAccounts',
         component: () => import('../components/adminSettings/LinkAccountsView.vue'),
-        meta: { requireAdmin: true },
+        meta: {
+          permissions: [PERMISSIONS.SITE_ADMIN],
+        },
       },
       {
         path: 'settings',
         name: 'ProfileSettings',
         component: () => import('../components/adminSettings/Settings.vue'),
+        meta: {
+          permissions: [],
+        },
       },
     ],
-    meta: { pageTitle: 'Profile' },
+    meta: {
+      pageTitle: 'Profile',
+      permissions: [],
+    },
   },
   {
     path: '/enable-cookies',
     name: 'EnableCookies',
     component: () => import('../pages/EnableCookies.vue'),
-    meta: { pageTitle: 'Enable Cookies' },
+    meta: {
+      pageTitle: 'Enable Cookies',
+      permissions: [],
+    },
   },
   {
     path: '/:pathMatch(.*)*',
     name: 'NotFound',
     component: () => import('../pages/NotFound.vue'),
-    meta: { pageTitle: 'Whoops! 404 Page!' },
+    meta: {
+      pageTitle: 'Whoops! 404 Page!',
+      permissions: [],
+    },
   },
   {
     path: '/add-users',
     name: 'Add Users',
     component: () => import('../pages/users/AddUsers.vue'),
-    meta: { pageTitle: 'Add Users', requireAdmin: true, project: 'LEVANTE' },
+    meta: {
+      pageTitle: 'Add Users',
+      permissions: [PERMISSIONS.SITE_ADMIN],
+    },
   },
 
   {
     path: '/link-users',
     name: 'Link Users',
     component: () => import('../pages/users/LinkUsers.vue'),
-    meta: { pageTitle: 'Link Users', requireAdmin: true, project: 'LEVANTE' },
+    meta: {
+      pageTitle: 'Link Users',
+      permissions: [PERMISSIONS.SITE_ADMIN],
+    },
   },
   // {
   //   path: '/edit-users',
   //   name: 'Edit Users',
   //   component: () => import('../pages/users/EditUsers.vue'),
-  //   meta: { pageTitle: 'Edit Users', requireAdmin: true, project: 'LEVANTE' },
+  //   meta: { permissions: [],  pageTitle: 'Edit Users', requireAdmin: true, project: 'LEVANTE' },
   // },
   {
     path: '/survey',
     name: 'Survey',
     component: () => import('../pages/UserSurvey.vue'),
-    meta: { pageTitle: 'Survey', project: 'LEVANTE' },
+    meta: {
+      pageTitle: 'Survey',
+      permissions: [],
+    },
   },
   {
     path: '/maintenance',
     name: 'Maintenance',
     component: () => import('../pages/MaintenancePage.vue'),
-    meta: { pageTitle: 'Down for Maintenance' },
+    meta: {
+      pageTitle: 'Down for Maintenance',
+      permissions: [],
```

Comment:
all the instances of permissions should be roles. Permissions is the actual actions they're allowed to take.

Reply:
maybe `allowedRoles` would be good to be descriptive

Reply:
or `allowedRouteRoles`